helix-view/src/document.rs

```diff
@@ -388,6 +399,15 @@ impl Document {
         }
 
         doc.detect_indent_and_line_ending();
+        // this seems overly complex but should make it easier
+        // to extend doc.file_stat with more information in the future.
+        doc.write_permission = fs::metadata(path).map_or(WritePermission::Unknown, |f| {
+            if f.permissions().readonly() {
+                WritePermission::ReadOnly
+            } else {
+                WritePermission::Writable
+            }
+        });
```

Comment:
Should probably be part of `doc.set_path` since a path can also be changed after opening?

Reply:
What are the implications of changing a path after opening?